src/ser.rs

```diff
@@ -78,7 +84,9 @@ impl<'py> Serialize for SerializePyObject<'py> {
 
             for (k, v) in py_dict {
                 let v_ob_type = v.get_type_ptr() as usize;
-                if v_ob_type == lookup.dict {
+                if v_ob_type == lookup.none && self.omit_none {
```

Comment:
```suggestion
                if self.omit_none && (v_ob_type == lookup.none || k.is_none()) {
```
I guess `self.omit_none` is more likely to be false.